Mention that we only support standard androidx.room setup (#4399)
getsentry/sentry-docs: src/platforms/android/common/performance/instrumentation/automatic-instrumentation.mdx

</Note>

> The [Sentry Android Gradle Plugin](/platforms/android/proguard/#gradle-configuration) does the tracing auto instrumentation via bytecode manipulation for `androidx.sqlite` and `androidx.room` libraries.

Reviewer: ```suggestion
The [Sentry Android Gradle Plugin](/platforms/android/proguard/#gradle-configuration) does the tracing auto instrumentation using bytecode manipulation for `androidx.sqlite` and `androidx.room` libraries.
```

The Plugin injects a code snippet that starts a span out of the active span bound to the scope for each `CRUD` operation. The SDK sets the span `operation` to `db` and `description` to the SQL Query if available.

The span finishes once the operation has been executed. The span `status` is set to `SpanStatus.OK` if successful or `SpanStatus.INTERNAL_ERROR` if there was any error.

When the operation throws an `Exception`, Sentry's SDK associates this exception to the running span. If you haven't set the SDK to swallow the exception and capture it, the span and SentryEvent will be linked when viewing it on the **Issue Details** page in sentry.io.

<Note>
 

> At the moment, only a standard `androidx.room` usage is supported, i.e. SQL queries for any other `SupportSQLiteOpenHelper.Factory` than [androidx.sqlite](https://github.com/androidx/androidx/tree/androidx-main/sqlite) will not be reported. However, it's worth reporting an [issue](https://github.com/getsentry/sentry-android-gradle-plugin/issues), as we could consider adding a new integration.

Reviewer 2: It would be nice to use the active voice to state the subject clearly. When using the passive voice, you can leave out the subject, and you don't have to be specific. I think it is essential to be clear when writing docs.

Author: I am not sure what exactly you are referring to?

Reviewer: To @philipphofmann's first point, I've changed the first sentence to active voice. So I'm stating that "we (Sentry)" only support standard usage, rather than it not being clear who is or isn't supporting standard usage. I don't know if this is technically accurate or not but just demonstrating how that might look. Having said that, sometimes there are good use cases for passive voice. I'm not sure if the "who" in the second sentence is the SDK or something else, so I don't know if it makes sense to change that sentence or not. 

```suggestion
At the moment, we only support standard `androidx.room` usage. For example, SQL queries for any `SupportSQLiteOpenHelper.Factory` other than [androidx.sqlite](https://github.com/androidx/androidx/tree/androidx-main/sqlite) will not be reported. However, it's worth reporting an [issue](https://github.com/getsentry/sentry-android-gradle-plugin/issues), as we might consider adding a new integration.
```

Author: Aha, that's what was meant, gotcha. Thanks for the suggestion 👍

Author: yeah in the 2nd sentence it's the SDK "SDK will not report SQL queries for any `blah` other than ....`

> At the moment, only a standard `androidx.room` usage is supported, i.e. SQL queries for any other `SupportSQLiteOpenHelper.Factory` than [androidx.sqlite](https://github.com/androidx/androidx/tree/androidx-main/sqlite) will not be reported. However, it's worth reporting an [issue](https://github.com/getsentry/sentry-android-gradle-plugin/issues), as we could consider adding a new integration.

Reviewer 2: I don't fully understand what you mean by

> However, it's worth reporting an issue, as we could consider adding a new integration.

For what should I report an issue?

Author: Hm, do you understand the previous sentence?

Reviewer: I'm also struggling with this last sentence. We've told the users about a limitation. Is this just a general appeal for people to tell us about problems with this plugin? Maybe this would be a better approach?
```suggestion
At the moment, only a standard `androidx.room` usage is supported, i.e. SQL queries for any other `SupportSQLiteOpenHelper.Factory` than [androidx.sqlite](https://github.com/androidx/androidx/tree/androidx-main/sqlite) will not be reported. If you experience any problems using this plugin, please report the [issue](https://github.com/getsentry/sentry-android-gradle-plugin/issues), as we might consider adding a new integration.
```

Author: Alright, my wording is bs then. What I meant is not a general appeal, but something like "If you are using a different `SupportSQLiteOpenHelper.Factory`, please report an issue on Github, so we are aware and can possibly tackle this"

> At the moment, we only support standard `androidx.room` usage. For example, SQL queries for any `SupportSQLiteOpenHelper.Factory` other than [androidx.sqlite](https://github.com/androidx/androidx/tree/androidx-main/sqlite) will not be reported. However, it's worth reporting an [issue](https://github.com/getsentry/sentry-android-gradle-plugin/issues), as we might consider adding a new integration.

Reviewer: Okay how about this (also, apologies, I replaced i.e., with "for example" instead of "in other words/that is"). 

```suggestion
At the moment, we only support standard `androidx.room` usage. That is, the SDK will not report any SQL queries for any `SupportSQLiteOpenHelper.Factory` other than [androidx.sqlite](https://github.com/androidx/androidx/tree/androidx-main/sqlite). However, if you are using a different `SupportSQLiteOpenHelper.Factory`, please report an [issue on Github](https://github.com/getsentry/sentry-android-gradle-plugin/issues), so we are aware and can possibly work on it.
```

Author: that looks perfect, thanks @imatwawana

Author: And thanks @philipphofmann for pointing out

> </Note>

Reviewer 3: looks like the code highlight is not working for `Note`, See https://sentry-docs-git-enh-sqlite-instrumentation-clarificaiton.sentry.dev/platforms/android/performance/instrumentation/automatic-instrumentation/

Reviewer 3: no idea, looks correct.

Reviewer: I'm also stumped. Once we're done with the wording changes, I can pull in your branch and try to play with it.

Author: Maybe it's not supported in 2 adjacent Notes? not sure lol

> At the moment, only a standard `androidx.room` usage is supported, i.e. SQL queries for any other `SupportSQLiteOpenHelper.Factory` than [androidx.sqlite](https://github.com/androidx/androidx/tree/androidx-main/sqlite) will not be reported. It's worth reporting an issue though, so we can add a new integration, check [this issue](https://github.com/getsentry/sentry-android-gradle-plugin/issues/234) for more details.

Author: ```suggestion
At the moment, only a standard `androidx.room` usage is supported, i.e. SQL queries for any other `SupportSQLiteOpenHelper.Factory` than [androidx.sqlite](https://github.com/androidx/androidx/tree/androidx-main/sqlite) will not be reported. It's worth reporting an issue though, as we can consider adding a new integration.
```
